pkg/asset/cluster/cluster.go

```diff
@@ -82,7 +84,7 @@ func (c *Cluster) Generate(parents map[asset.Asset]*asset.State) (*asset.State,
 	stateFile, err := terraform.Apply(tmpDir)
 	if err != nil {
 		// we should try to fetch the terraform state file.
-		log.Errorf("terraform failed: %v", err)
+		logrus.Errorf("Terraform failed: %v", err)
```

Comment:
So the `metadata` asset requires cluster and installconfig as the plan was to include some useful information like master ips etc in `clustermeta.json` file.

If we return error here the store wouldn't run generate on Metadata asset, which we might need in case of partial terraform error to destroy. We might wanna move metadata.json to this asset then. ??

Reply:
@wking I don't understand what this code is trying to do.

Reply:
> I don't understand what this code is trying to do.

The old code (and my suggestion above) are trying to preserve the Terraform state even if `apply` fails.  I think we want to keep that goal, but improve the implementation.

Reply:
Ahhh, gotcha.